src/xma/src/xmaapi/xma_utils.cpp

```diff
@@ -28,11 +29,79 @@
 #include <boost/filesystem/path.hpp>
 #include <iostream>
 #include <sstream>
+#include <bitset>
 
 #define XMAUTILS_MOD "xmautils"
 
 extern XmaSingleton *g_xma_singleton;
 
+namespace xma_core {
+    std::map<XmaSessionType, const char*> sessionMap = {
+        { XmaSessionType::XMA_SCALER, "scaler"},
+        { XmaSessionType::XMA_ENCODER, "encoder"},
+        { XmaSessionType::XMA_DECODER, "decoder"},
+        { XmaSessionType::XMA_FILTER, "filter"},
+        { XmaSessionType::XMA_KERNEL, "kernel"},
+        { XmaSessionType::XMA_ADMIN, "admin"},
+        { XmaSessionType::XMA_INVALID, "invalid"}
+    };
+
+    int32_t finalize_ddr_index(XmaHwKernel* kernel_info, int32_t req_ddr_index, int32_t* ddr_index, char* prefix) {
+        *ddr_index = -1;
+        if (kernel_info->soft_kernel) {
+            if (req_ddr_index != 0) {
+                xma_logmsg(XMA_WARNING_LOG, prefix, "XMA session with soft_kernel only allows ddr bank of zero\n");
+            }
+            //Only allow ddr_bank == 0;
+            *ddr_index = 0;
+            xma_logmsg(XMA_DEBUG_LOG, prefix, "XMA session with soft_kernel default ddr_bank: %d\n", *ddr_index);
+            return XMA_SUCCESS;
+        }
+        if (req_ddr_index < 0) {
```

Comment:
Going with the previous them, this line becomes:

`if (req_ddr_index == INVALID_INDEX) {`

Reply:
keeping same for more flexibility